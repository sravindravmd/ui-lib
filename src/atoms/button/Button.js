import React from 'react'
import PropTypes from 'prop-types'
import { StyledButton } from './Button.styles'

export const Button = ({ text, ...rest }) => (
  <StyledButton {...rest}>{text}</StyledButton>
)

Button.defaultProps = {
  disabled: false,
  isPrimary: true,
  size: 'large',
  theme: 'none',
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  isPrimary: PropTypes.bool,
  size: PropTypes.oneOf(['small', 'large']),
  theme: PropTypes.oneOf(['none', 'dark', 'light']),
}
