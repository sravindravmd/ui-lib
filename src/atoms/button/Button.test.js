import React from 'react'
import renderer from 'react-test-renderer'
import { Button } from './Button'

describe('Button', () => {
  it('renders primary Button correctly with default props', () => {
    const fn = jest.fn()
    const tree = renderer
      .create(
        <Button
          text='Button Label'
          isPrimary={true}
          size='small'
          onClick={fn}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders primary Button correctly with size:small, theme: dark, isPrimary:false', () => {
    const fn = jest.fn()
    const tree = renderer
      .create(
        <Button
          text='Button Label'
          isPrimary={true}
          size='small'
          onClick={fn}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders primary Button correctly with size:large, theme: light, isPrimary:false', () => {
    const fn = jest.fn()
    const tree = renderer
      .create(
        <Button
          text='Button Label'
          isPrimary={false}
          size='large'
          theme='light'
          onClick={fn}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
