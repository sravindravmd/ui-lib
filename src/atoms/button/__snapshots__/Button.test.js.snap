// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders primary Button correctly with default props 1`] = `
<button
  className="sc-bdfBwQ lmuDVK"
  disabled={false}
  onClick={[MockFunction]}
  size="small"
>
  Button Label
</button>
`;

exports[`Button renders primary Button correctly with size:large, theme: light, isPrimary:false 1`] = `
<button
  className="sc-bdfBwQ QcMRY"
  disabled={false}
  onClick={[MockFunction]}
  size="large"
>
  Button Label
</button>
`;

exports[`Button renders primary Button correctly with size:small, theme: dark, isPrimary:false 1`] = `
<button
  className="sc-bdfBwQ lmuDVK"
  disabled={false}
  onClick={[MockFunction]}
  size="small"
>
  Button Label
</button>
`;
